src/timearray.jl

```diff
@@ -15,17 +15,15 @@ immutable TimeArray{T, N, D<:TimeType, A<:AbstractArray} <: AbstractTimeSeries
                        values::AbstractArray{T,N},
                        colnames::Vector{String},
                        meta::Any)
-                           nrow, ncol = size(values, 1), size(values, 2)
-                           nrow != size(timestamp, 1) ? error("values must match length of timestamp"):
-                           ncol != size(colnames,1) ? error("column names must match width of array"):
-                           # length(unique(colnames)) != ncol ? colnames = replace_dupes(colnames):
-                           timestamp != unique(timestamp) ? error("there are duplicate dates"):
-                           ~(flipdim(timestamp, 1) == sort(timestamp) || timestamp == sort(timestamp)) ? error("dates are mangled"):
-                           flipdim(timestamp, 1) == sort(timestamp) ?
-                           new(flipdim(timestamp, 1), flipdim(values, 1), replace_dupes(colnames), meta):
-                           #new(flipdim(timestamp, 1), flipdim(values, 1), colnames, meta):
-                           new(timestamp, values, replace_dupes(colnames), meta)
-                           #new(timestamp, values, colnames, meta)
+        nrow, ncol = size(values, 1), size(values, 2)
+        nrow != size(timestamp, 1) ? error("values must match length of timestamp"):
+        ncol != size(colnames,1) ? error("column names must match width of array"):
+        # length(unique(colnames)) != ncol ? colnames = replace_dupes(colnames):
+        timestamp != unique(timestamp) ? error("there are duplicate dates"):
+        ~(flipdim(timestamp, 1) == sort(timestamp) || timestamp == sort(timestamp)) ? error("dates are mangled"):
+        flipdim(timestamp, 1) == sort(timestamp) ?
+        new(flipdim(timestamp, 1), flipdim(values, 1), replace_dupes(colnames), meta):
+        new(timestamp, values, replace_dupes(colnames), meta)
```

Comment:
Can I rewrite this single huge conditional ternary statement into if statements for readibility? Any effect on performance if we use `if` ?